binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java

```diff
@@ -49,123 +48,130 @@ abstract class SerdeResolverUtils {
 		}
-		else if (Short.class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.Short();
+
+		// Finally, just resort to using the fallback
+		return fallbackSerde;
+	}
+
+	private static Serde<?> getStandardDefaultSerde(Class<?> genericRawClazz) {
+		if (Integer.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.Integer();
 		}
-		else if (Double.class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.Double();
+		if (Long.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.Long();
 		}
-		else if (Float.class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.Float();
+		if (Short.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.Short();
 		}
-		else if (byte[].class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.ByteArray();
+		if (Double.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.Double();
 		}
-		else if (String.class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.String();
+		if (Float.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.Float();
 		}
-		else if (UUID.class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.UUID();
+		if (byte[].class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.ByteArray();
 		}
-		else if (!isSerdeFromStandardDefaults(fallbackSerde)) {
-			// User purposely set a default serde that is not one of the above
-			serde = fallbackSerde;
+		if (String.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.String();
 		}
-		else {
-			// If the type is Object, then skip assigning the JsonSerde and let the fallback mechanism takes precedence.
-			if (!genericRawClazz.isAssignableFrom((Object.class))) {
-				serde = new JsonSerde(genericRawClazz);
-			}
+		if (UUID.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.UUID();
 		}
-		return serde;
+		return null;
 	}
 
 	private static boolean isSerdeFromStandardDefaults(Serde<?> serde) {
-		if (serde != null) {
-			if (Number.class.isAssignableFrom(serde.getClass())) {
-				return true;
-			}
-			else if (Serdes.ByteArray().getClass().isAssignableFrom(serde.getClass())) {
-				return true;
-			}
-			else if (Serdes.String().getClass().isAssignableFrom(serde.getClass())) {
-				return true;
-			}
-			else if (Serdes.UUID().getClass().isAssignableFrom(serde.getClass())) {
-				return true;
-			}
-		}
-		return false;
+		if (serde == null) {
+			return false;
+		}
+		Class<?> serdeClass = serde.getClass();
+		return Serdes.Integer().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.Long().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.Short().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.Double().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.Float().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.ByteArray().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.String().getClass().isAssignableFrom(serdeClass) ||
+			Serdes.UUID().getClass().isAssignableFrom(serdeClass); //NOSONAR
```

Comment:
I think this `//NOSONAR` has to be on a first line of the "failed" code block:
https://github.com/spring-projects/spring-integration/blob/main/spring-integration-http/src/main/java/org/springframework/integration/http/multipart/UploadedMultipartFile.java#L67 - direct array argument storage.
thttps://github.com/spring-projects/spring-integration/blob/main/spring-integration-http/src/main/java/org/springframework/integration/http/multipart/UploadedMultipartFile.java#L67 - too long method.

Reply:
Got it - I thought it was at the end of the statement. I will adjust and then if ti fails during Jenkins I will adjust that.

Reply:
Yeah... I'm not sure there is SonarCube involved in Spring Cloud Stream CI plan.
@sobychacko ?

Reply:
So, @onobc , are you going to move that `//NOSONAR`?
Or let it just live as is?

Reply:
Yes, @artembilan, there is one on our Jenkins CI plans.

Reply:
Moving, give me a second. Plus, now I am tweaking around w/ those methods. I will commit shortly.